bridge-common/src/result_types.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prover::EthAddress;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{Balance, BlockHeight};
+
+pub type ResultPrefix = [u8; 32];
+
+pub const RESULT_PREFIX_WITHDRAW: ResultPrefix = [
+    246, 181, 82, 1, 204, 142, 13, 135, 141, 218, 111, 185, 98, 41, 84, 186, 141, 37, 252, 127, 56,
+    255, 227, 34, 202, 249, 14, 225, 115, 248, 131, 171,
+];
+pub const RESULT_PREFIX_LOCK: ResultPrefix = [
+    10, 158, 184, 119, 69, 133, 121, 219, 206, 131, 234, 87, 213, 86, 190, 80, 209, 195, 22, 11,
+    181, 241, 113, 159, 177, 114, 189, 51, 0, 172, 134, 35,
+];
+pub const RESULT_PREFIX_METADATA: ResultPrefix = [
+    179, 21, 212, 214, 232, 242, 53, 245, 250, 187, 11, 26, 15, 17, 133, 7, 246, 200, 84, 47, 174,
+    142, 26, 149, 102, 171, 230, 7, 98, 4, 124, 22,
+];
+
+#[derive(Debug, Eq, PartialEq, BorshSerialize, BorshDeserialize)]
+pub struct Withdraw {
+    #[cfg(feature = "result_with_prefix")]
+    pub prefix: ResultPrefix,
+    pub amount: Balance,
+    pub token: EthAddress,
+    pub recipient: EthAddress,
+}
+
+impl Default for Withdraw {
+    fn default() -> Self {
+        Self {
+            #[cfg(feature = "result_with_prefix")]
+            prefix: RESULT_PREFIX_WITHDRAW,
+            amount: Default::default(),
+            token: Default::default(),
+            recipient: Default::default(),
+        }
+    }
+}
```

Comment:
It's easy to rely on that everyone will use the `Default` trait. What we actually need is to implement some kind of construct so it's always enforced to use that specific prefix value for each of the types.

Reply:
Fixed 9b681b9059f63682bab63dfc29dfe768d7997916